go/adbc/driver/bigquery/driver.go

```diff
@@ -0,0 +1,222 @@
+	OptionStringQueryPriority          = "adbc.bigquery.sql.query.priority"
+	OptionIntQueryMaxBillingTier       = "adbc.bigquery.sql.query.max_billing_tier"
+	OptionIntQueryMaxBytesBilled       = "adbc.bigquery.sql.query.max_bytes_billed"
+	OptionBoolQueryUseLegacySQL        = "adbc.bigquery.sql.query.use_legacy_sql"
+	OptionBoolQueryDryRun              = "adbc.bigquery.sql.query.dry_run"
+	OptionBoolQueryCreateSession       = "adbc.bigquery.sql.query.create_session"
+	OptionIntQueryJobTimeout           = "adbc.bigquery.sql.query.job_timeout"
+)
+
+var (
+	infoVendorVersion string
+)
+
+func init() {
+	if info, ok := debug.ReadBuildInfo(); ok {
+		for _, dep := range info.Deps {
+			switch {
+			case dep.Path == "cloud.google.com/go/bigquery":
+				infoVendorVersion = dep.Version
+			}
+		}
+	}
+}
+
+type driverImpl struct {
+	driverbase.DriverImplBase
+	alloc memory.Allocator
+}
+
+// NewDriver creates a new BigQuery driver using the given Arrow allocator.
+func NewDriver(alloc memory.Allocator) adbc.Driver {
+	info := driverbase.DefaultDriverInfo("BigQuery")
+	if infoVendorVersion != "" {
+		if err := info.RegisterInfoCode(adbc.InfoVendorVersion, infoVendorVersion); err != nil {
+			panic(err)
+		}
+	}
+	return driverbase.NewDriver(&driverImpl{
+		DriverImplBase: driverbase.NewDriverImplBase(info, alloc),
+		alloc:          alloc,
+	})
+}
+
+func (d *driverImpl) NewDatabase(opts map[string]string) (adbc.Database, error) {
+	opts = maps.Clone(opts)
+	db := &databaseImpl{
+		DatabaseImplBase: driverbase.NewDatabaseImplBase(&d.DriverImplBase),
+		alloc:            d.alloc,
+		authType:         OptionValueAuthTypeDefault,
+	}
+	if err := db.SetOptions(opts); err != nil {
+		return nil, err
+	}
+
+	return driverbase.NewDatabase(db), nil
+}
+
+func stringToBool(value string) (bool, error) {
+	switch value {
+	case "true":
+		return true, nil
+	case "false":
+		return false, nil
+	default:
+		return false, adbc.Error{
+			Code: adbc.StatusInvalidArgument,
+			Msg:  fmt.Sprintf("Invalid boolean value, expected `true` or `false`, got `%s`", value),
+		}
+	}
+}
+
+func stringToTable(value string) (*bigquery.Table, error) {
+	parts := strings.Split(value, ".")
+	if len(parts) != 3 {
+		return nil, adbc.Error{
+			Code: adbc.StatusInvalidArgument,
+			Msg:  fmt.Sprintf("Invalid Table Reference format, expected `ProjectId.DatasetId.TableId`, got: `%s`", value),
+		}
+	}
+
+	return &bigquery.Table{
+		ProjectID: parts[0],
+		DatasetID: parts[1],
+		TableID:   parts[2],
+	}, nil
+}
+
+func stringToTableCreateDisposition(value string) (bigquery.TableCreateDisposition, error) {
+	switch value {
+	case "CREATE_NEVER":
+		return bigquery.CreateNever, nil
+	case "CREATE_IF_NEEDED":
+		return bigquery.CreateIfNeeded, nil
+	default:
+		return bigquery.CreateIfNeeded, adbc.Error{
+			Code: adbc.StatusInvalidArgument,
+			Msg:  fmt.Sprintf("unknown table create disposition value `%s`", value),
+		}
+	}
+}
```

Comment:
where is this used? can you just compare against the `CreateIfNeeded` and `CreateNever` constants instead of repeating them here?

```go
switch v := bigquery.TableCreateDisposition(value) {
case bigquery.CreateIfNeeded, bigquery.CreateNever:
        return v, nil
default:
        return v, adbc.Error{.....}
```